eng/scripts/SetTestPipelineVersion.ps1

```diff
@@ -0,0 +1,39 @@
+# Overides the project file and CHANGELOG.md for the template project using the next publishable version
+# This is to help with testing the release pipeline.
+. "${PSScriptRoot}\..\common\scripts\common.ps1"
+$latestTags = git tag -l "azure-template*"
+$semVars = @()
+
+$versionFile = "${PSScriptRoot}\..\..\sdk\template\azure-template\azure\template\_version.py"
+$changeLogFile = "${PSScriptRoot}\..\..\sdk\template\azure-template\CHANGELOG.md"
+
+Foreach ($tags in $latestTags)
+{
+  $semVars += $tags.Replace("azure-template_", "")
+}
+
+$semVarsSorted = [AzureEngSemanticVersion]::SortVersionStrings($semVars)
+LogDebug "Last Published Version $($semVarsSorted[0])"
+
+$newVersion = [AzureEngSemanticVersion]::ParseVersionString($semVarsSorted[0])
+$newVersion.IncrementAndSetToPrerelease()
+LogDebug "Version to publish [ $($newVersion.ToString()) ]"
+
+$versionFileContent = Get-Content -Path $versionFile
+$newVersionFile = @()
+
+Foreach ($line in $versionFileContent)
+{
+    if($line.StartsWith("VERSION"))
+    {
+        $line = 'VERSION = "{0}"' -F $newVersion.ToString()
+    }
+    $newVersionFile += $line
+}
+
+Set-Content -Path $versionFile -Value $newVersionFile
```

Comment:
We should use the python versioning and change log update scripts here instead of re-implementing it here.

Reply:
We should at least use the change log operations so that we an be sure we have one place for updating this logic.

Reply:
I cannot use those scripts because they depend on the already existing version in the file. In this case we are trying to set a new version based on the latest pushed tag. It does not depend on the version currently in source, this is more like a forced overwrite of the version rather than a bump.

Reply:
Another problem is the changelog logic has no ability to add content to the release note entry.

Reply:
>Another problem is the changelog logic has no ability to add content to the release note entry. 

That seems like a useful addition to the change log operations to me.

Reply:
I can do that as a follow up work.

Reply:
I'm fine with a follow-up but please file a tracking issue as I'd like to make sure we don't have these extra places to maintain.